app/assets/src/components/common/DetailsSidebar/GeneDetailsMode/GeneDetailsMode.jsx

```diff
@@ -0,0 +1,430 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { getCARDIndex, getAroEntry } from "~/api/amr";
+
+import cs from "./gene_details_mode.scss";
+
+const SOURCE_CARD = "card";
+const SOURCE_PUBMED = "pubmed";
+const SOURCE_GOOGLE_SCHOLAR = "googlescholar";
+const SOURCE_NCBI_REF_GENE = "ncbirefgene";
+
+const URL_CARD_ARO = "https://card.mcmaster.ca/aro/";
+const URL_PUBMED = "https://www.ncbi.nlm.nih.gov/pubmed/";
+const URL_GOOGLE_SCHOLAR = "https://scholar.google.com/scholar?q=";
+const URL_NCBI_REF_GENE =
+  "https://www.ncbi.nlm.nih.gov/pathogens/isolates#/refgene/";
+
+const CARD_FAMILY = "AMR Gene Family";
+const CARD_CLASS = "Drug Class";
+const CARD_MECHANISM = "Resistance Mechanism";
+const CARD_SYNONYMS = "Synonym(s)";
+const CARD_PUBLICATIONS = "Publications";
+
+// xml tags used in the aro owl xml file
+const XML_OWL_CLASSES = "owl:Class";
+const XML_OWL_LABEL = "rdfs:label";
+const XML_OWL_SYNONYM = "oboInOwl:hasExactSynonym";
+const XML_OWL_DESCRIPTION = "obo:IAO_0000115";
+const XML_OWL_ACCESSION = "oboInOwl:id";
+
+export default class GeneDetailsMode extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      cardEntryFound: false,
+      cardIndex: false,
+      collapseOntology: true,
+    };
+  }
+
+  componentDidMount() {
+    this.getGeneInfo(this.props.geneName);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.geneName !== prevProps.geneName) {
+      this.setState({
+        loading: true,
+        cardEntryFound: false,
+        collapseOntology: true,
+      });
+      this.getGeneInfo(this.props.geneName);
+    }
+  }
+
+  async getGeneInfo(geneName) {
+    const { cardIndex } = this.state;
+    const updatedOntology = {
+      accession: undefined,
+      label: "---",
+      synonyms: "---",
```

Comment:
I think it would look better just not to show the section for 'Synonym(s)' and these other fields if there's no info, instead of the `---`

Reply:
Done!